plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -0,0 +1,877 @@
+                    "label": "Prime Tower %",
+                    "description": "Enter the fan percentage for the Prime Tower (whether it's used or not).",
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_layer_or_feature == 'by_feature'"
+                },
+                "fan_bridge":
+                {
+                    "label": "Bridge %",
+                    "description": "Enter the fan percentage for any Bridging (whether it's used on not).",
+                    "type": "int",
+                    "default_value": 100,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_layer_or_feature == 'by_feature'"
+                },
+                "fan_combing":
+                {
+                    "label": "Fan 'OFF' during Combing:",
+                    "description": "When checked will set the fan to 0% for combing moves over 5 lines long in the gcode. When un-checked the fan speed during combing is whatever the previous speed is set to.",
+                    "type": "bool",
+                    "enabled": "fan_layer_or_feature == 'by_feature'",
+                    "default_value": true
+                },
+                "fan_feature_final":
+                {
+                    "label": "Final %",
+                    "description": "If you choose an 'End Layer' then this is the fan speed that will carry through to the end of the gcode file.  It will go into effect at the 'END' of your end layer.",
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "(int(fan_end_layer) != -1) and (fan_layer_or_feature == 'by_feature')"
+                },
+                "fan_enable_raft":
+                {
+                    "label": "Enable Raft Cooling",
+                    "description": "Enable the fan for the raft layers.  When enabled the Raft Fan Speed will continue until another Layer or Feature setting over-rides it.",
+                    "type": "bool",
+                    "default_value": false,
+                    "enabled": true
+                },
+                "fan_raft_percent":
+                {
+                    "label": "Raft Fan %:",
+                    "description": "Enter the percentage for the Raft.",
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_enable_raft"
+                }
+            }
+        }"""
+
+    def execute(self, data):
```

Comment:
I'd greatly prefer it if this main execute function was broken up into more sub-functions.

Reply:
I'll take another look.  I learn as I go.

Reply:
Alright.  I've removed the secondary classes and split 4 functions out of the long script.  It's still long but it's a little neater.

Reply:
I've got time to take a look now, but I noticed you closed a bunch of plugins. I hope I didn't come on too strong in the reviews?

Reply:
Nah.  I'll never be a software engineer and I'm not going to bother to try.  The only thing I know about a "lambda" is that it's a Greek letter.  The posts work for me and I'm good with that.
For something like the fan control it's been requested for over six years and I agreed it would be a really good idea so I gave it a shot.  I use it and DispalyInfoOnLCD on every print.

I'm sure all the UM code looks nice.  Then there are things like this [example](https://github.com/Ultimaker/Cura/issues/14764).  That bothered me on a very basic level.

Reply:
UM code is far from perfect. There are tons of things I'd change if I'd had the time (and energy).

It _is_ a really goof idea, and we'd like to have it in here. The other reason I made my comments the way they are is that in the past, you'd indicated that you'd like to learn.

I'm really sorry this is taking all so long -- ideally, that should not happen, but it is what it is. Next week the team will have a meeting about what to do about the slowness of the interaction with parts of our work outside of that what UM itself mandates (largely community focussed).

I'm not sure if I said it before (I hope I did), but none of this is really bad. I was debating myself whether to just merge it in wholesale without saying anything, but then I wouldn't have done due diligence.

I've finally rested up enough to prepare for work again. As I feel I have some unfinished business here, since I feel like I let you down, I'm going to try and make some small adjustments here and perhaps get it merged this weekend.